controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -326,27 +344,59 @@ func compareMachinesAndMembers(controlPlane *ControlPlane, members []*etcd.Membe
 				Reason:  controlplanev1.KubeadmControlPlaneMachineEtcdMemberNotHealthyV1Beta2Reason,
 				Message: fmt.Sprintf("Etcd doesn't have an etcd member for Node %s", machine.Status.NodeRef.Name),
 			})
+
+			// Instead, surface there is a machine without etcd member on kcp for the Available condition
+			// only if the machine is not deleting and the node exists by more than two minutes
+			// (this prevents the condition flick during scale up operations).
+			// Note: Two minutes is an arbitrary interval where we expect the system to detect the new etcd member on the machine.
+			if machine.DeletionTimestamp.IsZero() {
+				newNode := false
+				if nodes != nil {
+					for _, node := range nodes.Items {
+						if machine.Status.NodeRef.Name == node.Name && time.Since(node.CreationTimestamp.Time) < 2*time.Minute {
+							newNode = true
+						}
+					}
+				}
+				if !newNode {
+					membersAndMachinesAreMatching = false
+				}
+			}
 		}
 	}
 
 	// Check Etcd member -> Machine.
 	for _, member := range members {
 		found := false
+		hasProvisioningMachine := false
 		for _, machine := range controlPlane.Machines {
-			if machine.Status.NodeRef != nil && machine.Status.NodeRef.Name == member.Name {
+			if machine.Status.NodeRef == nil {
+				hasProvisioningMachine = true
+				continue
+			}
+			if machine.Status.NodeRef.Name == member.Name {
 				found = true
 				break
 			}
 		}
 		if !found {
+			// Surface there is an etcd member without a machine on into the EtcdClusterHealthy condition on kcp.
 			name := member.Name
 			if name == "" {
 				name = fmt.Sprintf("%d (Name not yet assigned)", member.ID)
 			}
-			kcpErrors = append(kcpErrors, fmt.Sprintf("Etcd member %s does not have a corresponding Machine", name))
+			if kcpErrors != nil {
+				*kcpErrors = append(*kcpErrors, fmt.Sprintf("Etcd member %s does not have a corresponding Machine", name))
+			}
+
+			// Instead, surface there is an etcd member without a machine on kcp for the Available condition
```

Comment:
```suggestion
			// Instead, surface there is an etcd member without a machine for the Available condition
```
not sure, "on KCP" sounds a bit confusing to me (similar below)

Reply:
improved the comment, PTAL